### PR TITLE
feat(postgres): implement `ops.Arbitrary`

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds support for the `.arbitrary()` method to the postgres backend.

### CODE REVIEW

**Comment by cpcloud** [comment]
Cloud backends look good:

```
cloud in 🌐 albatross in …/ibis on  postgres-arbitrary is 📦 v5.0.0 via 🐍 v3.10.10 via ❄️  impure (ibis-3.10.10-env)
❯ pytest -m 'bigquery or snowflake' -n auto --dist=loadgroup --snapshot-update --allow-snapshot-deletion -q
bringing up nodes...
xxxx.......x..xxx..x........xx..xx....x....x.xx..........x..............xxx...........x..x..........x..x...... [  5%]
.x...x.........xxx........xx..x.....xx.....................x.........x.......x.......x........................ [ 10%]
................................................x.................................................x....xxx.... [ 16%]
x.....x....xx..........xx.x.xx.x...x.x......x.....x...x...........x.....xx...xx..x......xxx..x.....x.......... [ 21%]
.........x..x.........x...x..x.......x.......x......x.x.x.x......x......xxxx.xxx...xx.x..x..xx..xx.xx.xxx...x. [ 27%]
...x.x.x.x.x....x....x..x.x.xxx.xxxxxxxxxx.xxxxxxx.....x..x....xxxx..........x....x........................... [ 32%]
x.................x........................x............xxx.......x....x........x........xx.x.xxx.....x....xx. [ 37%]
x.x...xxx.xx...x.xxx....x..x..x.x........x...xx....x.........xx.xxx...x.........xx.xx..x.............x........ [ 43%]
.....x...xx..x.xxx......xxx.xx......x.x.x.x........xx..xxx.xx....x.....xx.x..xxx......x....xx......x.......... [ 48%]
....x........xx..xx.xx.............x..........x..x.x.................x.....x.................................. [ 54%]
......x...x........x...x...x...............s..........................x.........x......................x...... [ 59%]
.x..................x..........x.............................x..............................x........x..x....x [ 65%]
x..............x........x.x....x...................xxxxxxxxxxxxx.....xs...x.xs....x.x....x..sx.xxx.x...x...... [ 70%]
............x........x.......x.................................x..........x.x..............................x.. [ 76%]
...........................................x.....x.x......x....x.xx....x.....x.....x........x.xxx.x..x.x...xx.. [ 81%]
.x..x......x.x..x...xx....xx..x...xx.x.x.x..x.x..........................x.............x...........x.......... [ 86%]
....................................................................................................x.......... [ 92%]
........................................x............................x....x.x.....x...........x...........x..x [ 97%]
.x...s.....x....................s.........                                                                     [100%]
1679 passed, 6 skipped, 339 xfailed in 201.09s (0:03:21)
```

File: ibis/backends/postgres/__init__.py [hunk 1]
```diff
@@ -115,6 +116,37 @@ def do_connect(
             alchemy_url, connect_args=connect_args, poolclass=sa.pool.StaticPool
         )
 
+        # define first/last aggs for ops.Arbitrary
+        #
+        # ignore exceptions so the rest of ibis still works: a user may not
```
[the remark this answers is not in the document]
Comment (jcrist) [reply ibis/backends/postgres/__init__.py:121]: Alternatively, could we (easily) delay registering the functions until the first time arbitrary is executed against a postgres connection? Afaict these functions persist after the connection, users that don't make use of them may not be enthused about the db being modified.

**Review by jcrist — COMMENTED** [review]
Overall this lgtm.

File: ibis/backends/postgres/__init__.py [hunk 2]
```diff
@@ -115,6 +116,37 @@ def do_connect(
             alchemy_url, connect_args=connect_args, poolclass=sa.pool.StaticPool
         )
 
+        # define first/last aggs for ops.Arbitrary
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/postgres/__init__.py:119]: Unfortunately, that doesn't seem to be deterministic. I'm not sure why, but under the connect handler the function sometimes get registered and sometimes they don't.

File: ibis/backends/postgres/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/postgres/__init__.py:121]: We can probably use one of the sqlalchemy before/after executes to create and drop aggregates before/after execution

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/postgres/__init__.py]: Fixed, these functions are created and dropped on every execute. This seems to have minimal effect on interactive latency locally.

**Review by kszucs — APPROVED** [review]

**Pull request merged**